src/ert/scheduler/job.py

```diff
@@ -167,6 +170,49 @@ async def _max_runtime_task(self) -> None:
         )
         self.returncode.cancel()
 
+    async def _verify_checksum(self, timeout: int = 120) -> None:
+        # Wait for job runpath to be in the checksum dictionary
+        runpath = self.real.run_arg.runpath
+        while runpath not in self._scheduler.checksum:
+            if timeout <= 0:
```

Comment:
This loop never runs in the tests. Should probably have a look at why that is. Should be something we either don't require or test for.

Reply:
Removing this will assume the checksum information reaches the scheduler once the job is complete. This is the case now but was not true when these checksum messages were batched in the ensemble evaluator `BatchingDispatcher`.

I want to keep it to guard against possible connectivity issues that might delay these messages. I will add a test that hits this loop as well

Reply:
:+1: